operator/pkg/gateway-api/helpers.go

```diff
@@ -178,3 +179,22 @@ func setMergedLabelsAndAnnotations(temp, desired client.Object) {
 	temp.SetAnnotations(mergeMap(temp.GetAnnotations(), desired.GetAnnotations()))
 	temp.SetLabels(mergeMap(temp.GetLabels(), desired.GetLabels()))
 }
+
+// hasCiliumManagedGateway checks if any of the route's parent refs point to a Cilium-managed gateway.
+// Returns true if at least one parent gateway is managed by Cilium, false otherwise.
+func hasCiliumManagedGateway(ctx context.Context, c client.Client, parentRefs []gatewayv1.ParentReference, getGatewayFn func(ref gatewayv1.ParentReference) (*gatewayv1.Gateway, error), logger *slog.Logger) (bool, error) {
+	for _, parent := range parentRefs {
+		if (parent.Group == nil || *parent.Group == gatewayv1.GroupName) &&
+			(parent.Kind == nil || *parent.Kind == kindGateway) {
+
+			gw, err := getGatewayFn(parent)
+			if err != nil {
+				return false, fmt.Errorf("failed to get Gateway: %w", err)
```

Comment:
@sayboras The end result of this line is that if there is no Gateway associated with a route, errors will be logged - I don't know to what extent this behavior is breaking existing integration tests and whether this particular failure mode should be a warning instead of an error.

Reply:
sorry for late reply, I am taking a look at this PR and get back to you soon.

Reply:
WDYT about expanding the predicate function ? 

https://github.com/cilium/cilium/blob/ac46da9fcc48dce1d5f9e7ca4d600aa8332622b1/operator/pkg/gateway-api/httproute.go#L131

Reply:
Just pushed a new set of commits that adopts that approach; will test locally and see if this is viable

Reply:
Looks like this method is working as intended:

`cilium-operator-86649d6bc8-xlk9r cilium-operator 2024-12-28T17:00:46.774617211-05:00 time=2024-12-28T22:00:46Z level=debug msg="Gateway does not have matching controller, skipping" module=operator.operator-controlplane.leader-lifecycle.gateway-api controller=gateway resource=default/non-cilium-httproute`

Reply:
thanks for your help investigating and confirming the result :heavy_check_mark: